src/DSDPSolverInterface.jl

```diff
@@ -0,0 +1,242 @@
+using SemidefiniteOptInterface
+SOI = SemidefiniteOptInterface
+
+using MathOptInterface
+MOI = MathOptInterface
+
+export DSDPSolver
+
+struct DSDPSolver <: SOI.AbstractSDSolver
+    options::Dict{Symbol,Any}
+end
+DSDPSolver(;kwargs...) = DSDPSolver(Dict{Symbol,Any}(kwargs))
+
+mutable struct DSDPSolverInstance <: SOI.AbstractSDSolverInstance
+    dsdp::DSDPT
+    lpcone::LPCone.LPConeT
+    nconstrs::Int
+    blkdims::Vector{Int}
+    blk::Vector{Int}
+    sdpcone::SDPCone.SDPConeT
+    nlpdrows::Int
+    lpdvars::Vector{Int}
+    lpdrows::Vector{Int}
+    lpcoefs::Vector{Cdouble}
+
+    x_computed::Bool
+    y_valid::Bool
+    y::Vector{Cdouble}
+    z_computed::Bool
+
+    options::Dict{Symbol,Any}
+    function DSDPSolverInstance(; kwargs...)
+        new(C_NULL, C_NULL, 0, Int[], Int[], C_NULL, 0, Int[], Int[], Cdouble[],
+            true, true, Cdouble[], true, Dict{Symbol, Any}(kwargs))
+    end
+end
+SOI.SDSolverInstance(s::DSDPSolver) = DSDPSolverInstance(; s.options...)
+
+function SOI.initinstance!(m::DSDPSolverInstance, blkdims::Vector{Int}, nconstrs::Int)
+    @assert nconstrs >= 0
+    m.nconstrs = nconstrs
+    m.blkdims = blkdims
+    m.nlpdrows = 0
+    sdpidx = 0
+    m.blk = similar(blkdims)
+    for i in 1:length(blkdims)
+        if blkdims[i] < 0
+            m.blk[i] = m.nlpdrows
+            m.nlpdrows -= blkdims[i]
+        else
+            sdpidx += 1
+            m.blk[i] = sdpidx
+        end
+    end
+    m.lpdvars = Int[]
+    m.lpdrows = Int[]
+    m.lpcoefs = Cdouble[]
+    m.dsdp = Create(nconstrs)
+    # TODO only create if necessary
+    m.sdpcone = CreateSDPCone(m.dsdp, length(blkdims))
+    for constr in 1:nconstrs
+        # TODO in examples/readsdpa.c line 162,
+        # -0.0 is used instead of 0.0 if the dual obj is <= 0., check if it has impact
```

Comment:
What would be the potential impact here? ``0.0 === -0.0`` in the library code?

Reply:
I don't think it could have any impact :-P I was just afraid that the library was checking for the sign if the Floating point number by looking at the binary representation in which cas it would have an impact. But I doubt it.